source/MaterialXRender/ImageHandler.cpp

```diff
@@ -208,4 +208,61 @@ void ImageHandler::clearImageCache()
     _imageCache.clear();
 }
 
+FilePath ImageHandler::getResolveUDIMInformation(const FilePath& filePath, const string& udimString, vector<int>& udimTile)
+{
+    FilePath resolvedFilePath = filePath;
+    udimTile.resize(2);
```

Comment:
Wasteful to allocate on the heap when we know we'll have two ints.

Reply:
At the very least, it should be std::pair<int, int> and a vector of these in the multiple case. It would also represent the pairs much more explicitly. But I would actually suggest defining a custom struct, something like
`struct UdimTile
{
    FilePath filePath;
   size_t row, column;
};`
- and return either one of or a vector of these. This would also get rid of the unintuitive output argument.